x-pack/platform/plugins/shared/streams/server/routes/content/route.ts

```diff
@@ -155,7 +186,40 @@ const importContentRoute = createServerRoute({
   },
 });
 
+const previewContentRoute = createServerRoute({
+  endpoint: 'POST /api/streams/{name}/content/preview 2023-10-31',
+  options: {
+    access: 'public',
+    summary: 'Preview a content pack',
+    description: 'Returns a json representation of a content pack.',
+    body: {
+      accepts: 'multipart/form-data',
+      maxBytes: MAX_CONTENT_PACK_SIZE_BYTES,
+      output: 'stream',
+    },
+  },
+  params: z.object({
+    path: z.object({
+      name: z.string(),
+    }),
+    body: z.object({
+      content: z.instanceof(Readable),
+    }),
+  }),
+  security: {
+    authz: {
+      enabled: false,
+      reason:
+        'This API delegates security to the currently logged in user and their Elasticsearch permissions.',
+    },
+  },
+  async handler({ params }) {
+    return await parseArchive(params.body.content);
```

Comment:
We can extract the archive in the client with [zip.js](https://gildas-lormeau.github.io/zip.js/). I opted for a server route instead to avoid introducing a new heavy library for this poc